main/ejml-core/src/org/ejml/concurrency/EjmlConcurrency.java

```diff
@@ -29,6 +30,9 @@
 	/** Used to toggle auto matic switching to concurrent algorithms */
 	public static boolean USE_CONCURRENT = true;
 
+	/** Minimum number of elements in a matrix before it will switch to concurrent implementation */
+	public static int ELEMENT_THRESHOLD = 50_000;
```

Comment:
Educated guess. I know for very small matrices it will be a big performance hit so this has to be added. It will definitely be operator dependent. I figured it wasn't worth delaying the feature for a more rigorous performance study that might not happen any time soon. Also people who use SimpleMatrix tend to not be super concerned about small hits to performance. Yes you are right there should be a study done to select this value.